File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    dict = {};
    getValue = (n) => {
        if(this.dict.hasOwnProperty(n)) {
            return this.dict[n];
        }
        return 0;
    }
    longestConsecutive(nums) {
        const n = nums.length;
        if(n === 0) {
            return 0;
        }
        let maxLen = 1;
        // LCS at i = LCS ending at i - 1 + 1 + LCS starting at i + 1
        for(const i of nums) {
            if(this.getValue(i) !== 0) {
                continue;
            }
                const lcs = this.getValue(i-1) + 1 + this.getValue(i+1);
                this.dict[i] = lcs;
                this.dict[i - this.getValue(i - 1)] = lcs;
                this.dict[i + this.getValue(i + 1)] = lcs;
                maxLen = Math.max(lcs, maxLen);
        }
        return maxLen;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-6.cs
public class Solution {
    public int FindMin(int[] nums) {
        int n = nums.Length;
        if(n == 1) {
            return nums[0];
        }   
        if(nums[0] < nums[n-1]) {
            return nums[0];
        }

        int i = 0, j = n-1;
        while(i <= j) {
            int mid = i + (j - i)/2;
            if(mid < j && nums[mid] > nums[mid + 1]) {
                return nums[mid + 1];
            }
            else if(mid > i && nums[mid] < nums[mid - 1]) {
                return nums[mid];
            }

            if(mid < j && nums[mid] < nums[j]) {
                j = mid - 1;
            } else if( mid > i && nums[mid] > nums[i]) {
                i = mid + 1;
            }
        }
        return -1;
    }
}
